Highlight current executed line (bg=red fg=white)

daltonmatos opened the pull request:
Hi,

this PR changes the way vim-vebugger highlights the current debugged line.

it now hightlitghs thw whole line, instead of adding a `->` sign on the line.


This is how it was berfore this PR:

![screenshot from 2017-10-16 11-56-26](https://user-images.githubusercontent.com/121638/31620169-52e3ae78-b275-11e7-8f9c-d449fe21927c.png)

This is how it is now, after the PR:

![screenshot from 2017-10-16 11-54-49](https://user-images.githubusercontent.com/121638/31620186-5d3b3594-b275-11e7-9806-c21bfc34c486.png)

idanarye commented:
I really hate the idea of using hard-coded colors here. [It's hard enough to make the builtin highlight groups work for every setup](https://github.com/idanarye/vim-vebugger/issues/13#issuecomment-93460511).

daltonmatos commented:
Makes sense. What do you think would be a good approach to have a better highlighted debugged line? Have config options?

Thanks,

daltonmatos commented:
Hello @idanarye,

I'm using gvim and even when I define `BreakPoint` hightlight-group, the current debugged line does not get highlighted, this small patch does the trick and the current debugged line gets correctly highlighted to whatever the `BreakPoint` is set to.

would you be OK merging if I update this PR?

```diff
diff --git a/autoload/vebugger/std.vim b/autoload/vebugger/std.vim
index 284bb7a..a54673f 100644
--- a/autoload/vebugger/std.vim
+++ b/autoload/vebugger/std.vim
@@ -304,6 +304,7 @@ sign define vebugger_current text=->
 
 if hlexists('BreakPoint')
     sign define vebugger_breakpoint text=** linehl=BreakPoint texthl=BreakPoint
+    sign define vebugger_current linehl=BreakPoint
 else
     sign define vebugger_breakpoint text=** linehl=ColorColumn texthl=ColorColumn
 endif
```
With this patch, the whole line is also highlighted if the `Breakpoint` highlight-group exist. I also think that checking for the `BreakPoint` is much better than having the colors hardcoded. This way I can define this group with whatever colors I want, without even having to change my colorscheme.

Also, when `BreakPoint` does not exist, vebugger remains with the same behavior.

Let me know what do you think.

idanarye commented:
`BreakPoint` is for breakpoints. If you want highlighting for the `vebugger_current` it should have a different highlight group.

daltonmatos commented:
Makes sense. What name do you suggest? `DebuggedLine` would be a good name?

idanarye commented:
Sounds good to me. Either that, or `ExecutedLine`.

daltonmatos committed:
Highlight current debugged line with DebuggedLine

If the `DebuggedLine` highlight-group exist, highlight the current
debugged line with it.

daltonmatos commented:
Done. Waiting for review.

Thanks,

idanarye commented on autoload/vebugger/std.vim:
```diff
@@ -302,6 +302,10 @@ endfunction
 
 sign define vebugger_current text=->
```
Mind moving this to an `else` clause?

daltonmatos replied:
Done.
